zerver/lib/alert_words.py

```diff
@@ -21,7 +21,7 @@ def add_user_alert_words(user_profile: UserProfile, alert_words: Iterable[str])
 
     new_words = [w for w in alert_words if w not in words]
     words.extend(new_words)
-
+    words.sort(key=len, reverse=True)
```

Comment:
I believe that this change and the corresponding test changes should be in a separate commit.